webaudio-audiocontext-close/index.html

```diff
@@ -0,0 +1,140 @@
+    <meta name="mobile-web-app-capable" content="yes">
+    <link rel="icon" sizes="192x192" href="../images/touch/chrome-touch-icon-192x192.png">
+
+    <!-- Add to homescreen for Safari on iOS -->
+    <meta name="apple-mobile-web-app-title" content="WebAudio AudioContext close Sample">
+
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-status-bar-style" content="black">
+    <link rel="apple-touch-icon-precomposed" href="../images/apple-touch-icon-precomposed.png">
+
+    <!-- Tile icon for Win8 (144x144 + tile color) -->
+    <meta name="msapplication-TileImage" content="images/touch/ms-touch-icon-144x144-precomposed.png">
+    <meta name="msapplication-TileColor" content="#3372DF">
+
+    <link rel="icon" href="../images/favicon.ico">
+
+    <link rel="stylesheet" href="../styles/main.css">
+  </head>
+
+  <body>
+    <h1>WebAudio AudioContext close Sample</h1>
+
+    <p>Available in <a href="http://www.chromestatus.com/feature/5411681005993984">Chrome 43+</a></p>
+
+    <p>
+      The <code>AudioContext.close</code> method closes the audio context, stopping audio operations, and releases any system resources in use.
+      It will not necessarily release objects <em>created</em> by this <code>AudioContext</code>.
+      This method returns a <code>Promise</code> that resolves when any resources blocking further creation of other audio contexts have been released.
+    </p>
+
+    <div class="output">
+      <div>
+        Manages a pool of AudioContext instances, allowing it to be grown or shrunk (by explicitly closing a random context).
+        If your browser prevents creation of further contexts, close one first.
+      </div>
+      <pre id="count"></pre>
+      <button id="btn-create">Create</button>
+      <button id="btn-close">Close</button>
+      <pre id="results" style="color: red"></pre>
+    </div>
+
+    <!-- // [START code-block] -->
+    <script>
+
+    var pool = [];
+    var i = 0;
+    var context;
+
+    // Create a large number of initial contexts. On Mac OS X running M44, this
+    // can create six contexts.
+    while (++i < 100) {
+      try {
+        context = new AudioContext();
+      } catch (e) {
+        break;  // give up, can't create more contexts
+      }
+      pool.push(context);
+    }
+
+    function refreshCountMessage(opt_message) {
+      count.innerText = 'pool: ' + pool.length + ' AudioContext instances';
+      results.innerText = opt_message || '';
+    }
+    refreshCountMessage()
+
+    function closeRandomContext() {
+      if (pool.length == 0) {
+        refreshCountMessage();
+        return;
+      }
+      var choice = Math.floor(Math.random() * pool.length);
+      var context = pool[choice];
+
+      var promise = context.close();
+      promise.then(function() {
+        pool.splice(pool.indexOf(context), 1);
+        refreshCountMessage('close promise resolved');
+      }, function(e) {
```

Comment:
Matter of personal preference, but (recently) instead of using the two-argument version of `then()`, I've found myself using a single-argument `then()` to handle the resolution followed by a `catch()` to handle the rejection.

Because it shares terminology with exceptions, developers who aren't familiar with promises will still get the idea as to what the `catch()` does, but seeing a two-argument `then()` doesn't give any hints.